api/src/main/proto/bio/terra/tanagra/underlay/search.proto

```diff
@@ -0,0 +1,81 @@
+// Messages describing how to construct Tanagra entity searches.
+
+syntax = "proto3";
+
+package bio.terra.tanagra.underlay;
+option java_package = "bio.terra.tanagra.proto.underlay";
+option java_multiple_files = true;
+
+import "bio/terra/tanagra/underlay/entities.proto";
+import "bio/terra/tanagra/underlay/mapping.proto";
+
+// The schema of how an entity filter can be constructed for a given entity.
+message EntityFiltersSchema {
+  // The name of the entity supported by this filter.
+  string entity = 1;
+
+  repeated FilterableAttribute attributes = 2;
+
+  repeated FilterableRelationship relationships = 3;
+}
+
+// An attribute on an entity that may be used for filtering with hints on how to find
+// interesting values.
+message FilterableAttribute {
+  // The attribute that can be used in the entity filter.
+  AttributeId attribute = 1;
+
+  // Hints are usually but not necessarily exclusive.
```

Comment:
I'm curious - what is the exception you are imagining here? An attribute which an overloaded purpose?

Reply:
... I guess I don't have an example. I can make it a oneof for now.